### PR TITLE
Added additional tests for `pip_parse` and `pip_install` utilties

### PR DESCRIPTION
**Pull request by UebelAndre**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Does not include precompiled binaries, eg. `.par` files. See [CONTRIBUTING.md](https://github.com/bazelbuild/rules_python/blob/master/CONTRIBUTING.md) for info
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature (please, look at the "Scope of the project" section in the README.md file)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [x] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

To close the loop on https://github.com/bazelbuild/rules_python/pull/519#issuecomment-909849717, tests have been added for the `data_requirement` and `dist_info_requirement` utilities of `pip_parse` and `pip_install`.

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

### CODE REVIEW

**Comment by UebelAndre** [comment]
The tests here make use of [py_test::env](https://docs.bazel.build/versions/4.0.0/be/common-definitions.html#test.env) which was released in Bazel `4.0.0`. The min tested version will need to be raised (392d3209aceef724daa99fedd79844dbef6c75ce).

**Comment by UebelAndre** [comment]
This PR is blocked by https://github.com/bazelbuild/rules_python/issues/526

File: .bazelversion [hunk 1]
```diff
@@ -1 +1 @@
-3.3.1
+4.0.0
```
Comment (alexeagle) [inline .bazelversion:1]: I don't think this belongs here?
If we want to bump our bazel version it should be in a separate PR

Comment (UebelAndre) [reply .bazelversion:1]: I commented that on the PR

https://github.com/bazelbuild/rules_python/pull/524#issuecomment-913681103

Comment (alexeagle) [reply .bazelversion:1]: I see. Would be easier for me to see them in the right order :)
Could you send that PR first and let's resolve that issue?

Comment (UebelAndre) [reply .bazelversion:1]: Opened https://github.com/bazelbuild/rules_python/pull/533

**Comment by UebelAndre** [comment]
This PR is now unblocked and ready for another review.

**Review by alexeagle — APPROVED** [review]
